src/lib/stores/bluetoothRead.ts

```diff
@@ -0,0 +1,145 @@
+import { everyEqual } from "$lib/array/everyEqual";
+import {
+  configurationPayloadTrailer,
+  radarDataOutputPayloadTrailer,
+} from "$lib/ld2410/constants";
+
+type ReadResult =
+  | { value: Uint8Array; done: false }
+  | { value: undefined; done: true };
+
+interface ReadEvent {
+  eventType: "READ";
+  payload: Uint8Array;
+}
+
+interface WriteEvent {
+  eventType: "WRITE";
+  payload: Uint8Array;
+}
+
+interface ConnectEvent {
+  eventType: "CONNECT";
+}
+
+interface SubscribedEvent {
+  eventType: "SUBSCRIBED";
+}
+
+interface DisconnectEvent {
+  eventType: "DISCONNECT";
+}
+
+type SerialEvent =
+  | ReadEvent
+  | WriteEvent
+  | ConnectEvent
+  | DisconnectEvent
+  | SubscribedEvent;
+
+type SubscribeCallback = (value: SerialEvent) => void;
+
+interface Store {
+  subscribe: (subscription: SubscribeCallback) => () => void;
+}
+
+export type SerialStore = Store & {
+  connect: () => void;
+  disconnect: () => void;
+  write: (payload: Uint8Array) => void;
+};
+
+export const createBluetoothReadStore = (
+  server: any
+): SerialStore => {
+  let stopping = false;
+  let sendPassCharacteristic: any | null = null;
+  let readDataCharacteristic: any | null = null;
+  const subs: SubscribeCallback[] = [];
+  const writeQueue: Uint8Array[] = [];
+
+  const broadcastEvent = (e: SerialEvent) => subs.forEach((cb) => cb(e));
+
+  const write = async (payload: Uint8Array) => {
+    writeQueue.push(payload);
+  };
+
+  const writeForever = async () => {
+    while (!stopping) {
+      if (writeQueue.length > 0) {
+
+        const payload = writeQueue.shift();
+        sendPassCharacteristic.writeValue(payload)
+
+        broadcastEvent({ eventType: "WRITE", payload });
+      }
+      await new Promise((x) => setTimeout(x, 100));
+    }
+  };
+
+  const readForever = async () => {
+    if (!readDataCharacteristic) { return }
+    await readDataCharacteristic.startNotifications();
+    readDataCharacteristic.addEventListener(
+      "characteristicvaluechanged",
+      (event: any) => {
+        broadcastEvent({
+          eventType: "READ",
```

Comment:
I'm seeing an odd bug where if I connect to bluetooth, disconnect, then reconnect, there's no output in the console component. It's always the second connection on bluetooth; serial doesn't have the same issue.

Looks like `outputDiv` in  `bind:this={outputDiv}` gets weirdly set to `null`. It's likely an existing bug and maybe even a bug from Svelte itself, so probably fine to leave a fix out of scope unless you have any ideas.